src/functions/keycloak/update-user-vacation/handler.ts

```diff
@@ -4,101 +4,113 @@ import type { APIGatewayProxyHandlerV2 } from "aws-lambda";
       body: JSON.stringify({ message: "User ID is required" }),
     };
   }
 
   try {
     const requestBody =
       typeof event.body === "string" ? JSON.parse(event.body) : event.body;
-    const { vacationDays } = requestBody;
 
-    if (!vacationDays) {
+    const { vacationDays } = requestBody as { vacationDays: VacationDays };
+
+    if (
+      !vacationDays ||
+      typeof vacationDays !== "object" ||
+      Array.isArray(vacationDays)
+    ) {
       return {
         statusCode: 400,
-        headers: corsHeaders,
         body: JSON.stringify({ message: "Vacation days are required" }),
       };
     }
 
     const userAttributes = await keycloakApiService.getUserAttributes(userId);
-    const updatedAttributes = { ...userAttributes };
 
     const vacationDaysByYear = userAttributes.vacationDaysByYear || [];
     const unspentVacationDaysByYear =
       userAttributes.unspentVacationDaysByYear || [];
 
-    // Store updated vacation days in a separate object
-    const updatedVacationDays: Record<
-      string,
-      { total: number; remaining: number }
-    > = {};
+    const currentYear = new Date().getFullYear();
+
+    const updatedVacationDays: Record<string, VacationDayEntry> = {};
+
+    const formatValue = (year: string, value: number) =>
+      `${year}:${String(value).padStart(3, "0")}`;
 
-    Object.entries(vacationDays).forEach(([year, data]: [string, any]) => {
+    const updateEntry = (arr: string[], value: string) => {
+      const index = arr.findIndex((v) => v.startsWith(`${value.split(":")[0]}:`));
+      if (index !== -1) arr[index] = value;
+      else arr.push(value);
+    };
+
+    Object.entries(vacationDays).forEach(([year, data]) => {
       const yearInt = Number.parseInt(year, 10);
-      if (!Number.isNaN(yearInt) && yearInt <= new Date().getFullYear()) {
-        const formattedTotal = `${year}:${String(data.total).padStart(3, "0")}`;
-        const formattedRemaining = `${year}:${String(data.remaining).padStart(
-          3,
-          "0"
-        )}`;
-
-        const updateEntry = (arr: string[], value: string) => {
-          const index = arr.findIndex((v) => v.startsWith(`${year}:`));
-          if (index !== -1) arr[index] = value;
-          else arr.push(value);
-        };
-
-        // Updating vacation days for the year
-        updateEntry(vacationDaysByYear, formattedTotal);
-        updateEntry(unspentVacationDaysByYear, formattedRemaining);
-
-        // Store the updated vacation days
-        updatedVacationDays[year] = {
-          total: data.total,
-          remaining: data.remaining,
-        };
-
-        updatedAttributes[`vacation_${year}`] = [String(data.total)];
-        updatedAttributes[`vacation_${year}_remaining`] = [
-          String(data.remaining),
-        ];
+      if (
+        Number.isNaN(yearInt) ||
+        yearInt > currentYear ||
+        typeof data.total !== "number" ||
+        typeof data.remaining !== "number"
+      ) {
+        return;
       }
```

Comment:
Are all of these checks really necessary?
Should there be some kind of log or error thrown if this happens?